pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -495,8 +505,6 @@ func testHTTPSClassification(t *testing.T, tr *Tracer, clientHost, targetHost, s
 				ctx.extras["cmd"] = cmd
 			},
 			postTracerSetup: func(t *testing.T, ctx testContext) {
-				cmd := ctx.extras["cmd"].(*exec.Cmd)
-				utils.WaitForProgramsToBeTraced(t, "shared_libraries", cmd.Process.Pid)
```

Comment:
the test is racy again, as you can see duplicated traffic (you can still hook the python server)

my biggest concern here is - this is the only test that uses a different hooking mechanism (shared libraries vs go).

What's the problem with the test? what is the failure we have?

reviewing the test - seems very awful test. I think we should try and make it work (and reduce what's going on here)
I do recommend the following - try to rewrite the test (with python server and shared_libraries hooking)

Reply:
> the test is racy again, as you can see duplicated traffic (you can still hook the python server)

You mean, we see the traffic from both sides because this?

My understanding is that this is also what happens when we run plaintext classification tests. In fact, I have had to modify a few tests to work with TLS, mainly by making sure the server sends a response. They were passing with plaintext because we were seeing all sides of it thanks to the socket filter.

> my biggest concern here is - this is the only test that uses a different hooking mechanism (shared libraries vs go).

> What's the problem with the test? what is the failure we have?

The process monitor does not always hook the Python server in time.

> reviewing the test - seems very awful test. I think we should try and make it work (and reduce what's going on here)
I do recommend the following - try to rewrite the test (with python server and shared_libraries hooking)

I agree. I would add that, like with the other tests, we should not be relying on the process monitor for these tests, and instead use the same approach as with the other TLS tests, by adding additional helpers to manually hook/unhook with shared libraries, like we did with GoTLS. (I would add those helpers in a separate PR)